store/sqlstore/channel_store.go

```diff
@@ -915,6 +915,41 @@ func (s SqlChannelStore) GetChannels(teamId string, userId string, includeDelete
 	return channels, nil
 }
 
+func (s SqlChannelStore) GetChannelsOpt(opt *model.GetChannelsOptions) (*model.ChannelList, *model.AppError) {
+	query := s.getQueryBuilder().
+		Select("c.*").
+		From("Channels AS c")
+
+	if len(opt.ChannelTypes) > 0 {
+		query = query.Where(fmt.Sprintf("c.Type IN ('%s')", strings.Join(opt.ChannelTypes, "', '")))
+	}
+
+	if len(opt.UserIds) > 0 {
+		query = query.Where(fmt.Sprintf("c.Id IN (SELECT DISTINCT(ChannelId) FROM ChannelMembers where UserId IN ('%s'))", strings.Join(opt.UserIds, "', '")))
+	}
+
+	if opt.Page != 0 {
+		query = query.Limit(uint64(opt.Page))
+	}
+
+	if opt.PerPage != 0 {
+		query = query.Offset(uint64(opt.PerPage))
+	}
```

Comment:
do you think it would be more appropriate to check this instead ? 
```suggestion
if opt.Page > 0 && opt.PerPage >= 0 {
	query = query.Limit(uint64(opt.Page)).Offset(uint64(opt.PerPage))
}
```

check if limit is greater than 0.